### ISSUE TITLE
Node ID occasionally disappears [MAKE LABEL POSITIONS EDITABLE]

### ISSUE THREAD

**Issue by tatsuro-ueda** [ArsMasiuk/qvge#41]
Maybe when a focus returns to the node, and when the node was just moved while zoom outing

**Comment by ArsMasiuk** [comment]
hello,
this could happen if some ids overlay one another.
in this case, one of them will be hidden to avoid wasting the space.
Maybe better solution would be to try to layout such ids differently...

Do you need to see all the ids at a time?

**Comment by weed** [comment]
OK, I get it. You mean that if density of node is very high, too many labels are shown and the appearance of graph isn't good.

I couldn't know this rule and felt that something doesn't work correctly. Maybe this is a subject of user experience. I suggest to add "show overlapped labels" to "view" menu. How about this?

**Comment by ArsMasiuk** [comment]
Exactly. The labels become not readable when they overlap. 
A quick solution would be, as you suggested, to add the menu entry "show overlapped labels".
A more sophisticated one would be, to introduce a kind of automatical "smart" layouting. I.e. shifting overlapping labels to optimize the space. But it is a quite complex task for now :(

**Comment by weed** [comment]
I agree with you. It is very complex. I have heard that there is a library of labeling on a digital map. It might be better to use that. 

Although smart layouting is not necessary immediately, I'd like to drag (move) labels. I show you a example as followings. Two nodes which are at the same height horizontally have to be arranged far from each other enough to show their entire labels. Sometimes I'd like to put a label upper or upper right-hand (or other sides) of a node.

![_ g__my drive_office_190331_2018 12 23 holomagnet5 gexf - qt visual graph editor 0 5 1 64bit 2018-12-30 11 28 03](https://user-images.githubusercontent.com/129954/50543869-8b7c9900-0c26-11e9-9d6a-0f2f06fab2fc.png)

**Comment by weed** [comment]
Should I create a new issue of this subject?

**Comment by weed** [comment]
For the moment, it's also not necessary to drag a label. It's enough to select the label position from options as "downer left-hand" "downer" "downer right-hand" "left" "right" "upper left-hand" "upper" "upper right-hand". That might be better because of simplicity.

**Comment by ArsMasiuk** [comment]
Ok, I got this. Yes, something like specifying the label position would be enough for the quick solution (like with ports). 
I think that a separate ticket is not necessary so we can stay on this one.

**Comment by weed** [comment]
All right. Anyway, I'm afraid that I might be a slave driver since it is in your vacation. (\*\_\*) (Is this English correct?)

**Comment by ArsMasiuk** [comment]
Oh, please don't excuse, your English is very good (and ways better than mine) :)
And I really appreciate all your comments because they are helping me drastically to improve the application's ux and logic. So I see your feedback as a very cooperative job, and I glad that my app is working :)

**Comment by weed** [comment]
I'm happy to hear that. I hope your success of development.

**Comment by ArsMasiuk** [comment]
An option is needed to let the label's text be always sticked to the bottom of the node (i.e. do not try to fit it automatically into the node box).

**Comment by ArsMasiuk** [comment]
A side bug: if node size > 15 then the text is drawn centred and sometimes disappears.

**Comment by ArsMasiuk** [comment]
For nodes, now it is possible to set the label position from the list.

**Comment by ArsMasiuk** [comment]
Icons added.
Not sure what to do with edge labels yet...
